src/app.service.js

```diff
@@ -335,4 +349,154 @@ export class AppService {
 
     return mainArticle.join(" ").trim().replace(/\\/g, "");
   }
+
+  gatherElementData(rootElement) {
+    const elementsData = [];
+
+    const traverse = (element, depth) => {
+      if (depth > 20 || !element) {
+        return;
+      }
+
+      const textLength = element.textContent.trim().length;
+      const allElements = element.querySelectorAll("*");
+      const linkElements = element.querySelectorAll("a");
+      const linkRatio = linkElements.length / (allElements.length + 1 || 1);
+
+      if (textLength === 0) {
+        return;
+      }
+
+      const elementData = {
+        element,
+        textLength,
+        linkRatio,
+      };
+
+      elementsData.push(elementData);
+
+      for (let i = 0; i < element.children.length; i += 1) {
+        traverse(element.children[i], depth + 1);
+      }
+    };
+
+    for (let i = 0; i < rootElement.children.length; i += 1) {
+      traverse(rootElement.children[i], 0);
+    }
+
+    return elementsData;
+  }
+
+  assignRelativeScores(elementsData) {
+    const maxTextLength = Math.max(
+      ...elementsData.map((data) => data.textLength),
+    );
+
+    elementsData.forEach((data) => {
+      const allElements = elementsData.length;
+      const childrenElements = data.element.childElementCount + 1;
+      const elementRatio = 1 - childrenElements / allElements;
+      const tagName = data.element.tagName.toLowerCase();
+      const className = data.element.className.toLowerCase();
+      const id = data.element.id.toLowerCase();
+      data.textLengthScore =
+        ((data.textLength * (1 - data.linkRatio)) / maxTextLength) * 100;
```

Comment:
```suggestion
      const allElements = elementsData.length;
      const childrenElements = data.element.childElementCount + 1;
      const elementRatio = 1 - childrenElements / allElements;
      const tagName = data.element.tagName.toLowerCase();
      const className = data.element.className.toLowerCase();
      const id = data.element.id.toLowerCase();
      
      data.textLengthScore =
        ((data.textLength * (1 - data.linkRatio)) / maxTextLength) * 100;
```

다른 코드와 일관된 헝태를 위해 개행 부탁드립니다.